docs/design/README.md

```diff
@@ -0,0 +1,595 @@
+#
+# [sdk-image]
+# registry = "public.ecr.aws/bottlerocket"
+# name = "bottlerocket-sdk"
+# version = "v0.30.0"
+
+# This is the container that Twoliter uses to invoke `cargo make` and other build
+# commands. It will be available for both x86_64 and Arm hosts, but it is agnostic
+# as to which architecture the build is targeting.
+[twoliter-image]
+registry = "public.ecr.aws/bottlerocket"
+name = "twoliter"
+version = "v0.1.0"
+```
+
+### Directory Structure
+
+Project directory structure is similar to what we see in Bottlerocket's main repo:
+
+```
+.
+├── build
+├── kits
+│   └── hello-dev-kit
+│       ├── src
+│       │   ├── kit.rs
+│       │   └── kit.spec
+│       └── Cargo.toml
+├── packages
+│   └── hello-agent
+│       ├── build.rs
+│       ├── Cargo.toml
+│       ├── hello-agent.service
+│       ├── hello-agent.spec
+│       └── pkg.rs
+├── sources
+│   ├── hello-agent
+│   │   ├── src
+│   │   │   └── main.rs
+│   │   └── Cargo.toml
+│   ├── Cargo.toml
+│   ├── clarify.toml
+│   └── deny.toml
+└── variants
+    ├── example-dev
+    │   ├── build.rs
+    │   ├── Cargo.toml
+    │   ├── pkg.rs
+    │   └── variant.spec
+    └── Cargo.toml
+```
+
+Directory descriptions:
+
+* build: The output directory for Bottlerocket images, RPM files, etc.
+* kits: Where the local kits are defined.
+* packages: Where the RPM packages are defined.
+* variants: Where Bottlerocket image builds are defined.
+
+### Twoliter New
+
+Inspired by `cargo new`, we will implement a `twoliter new` command that creates a new Twoliter project.
+The command creates a directory structure like the one shown above.
+The new project will include a minimal variant with a simple custom package and settings.
+The command will take a few arguments, such as `--project name` and `--variant-name`.
+
+Similarly, a command will be available to add a new variant to an existing project.
+For example, `twoliter new-variant` which will add a variant.
+
+```
+❯ twoliter new project --name ootb-variants --variant-name example-dev
+❯ cd ootb-variants
+❯ twoliter build variant --name example-dev --arch x86_64
+  ...
+❯ twoliter new variant --name my-new-variant
+```
+
+### Twoliter Update
+
+**TODO** - This section is a bit hand-wavy and will have its design further refined.
+See https://github.com/bottlerocket-os/twoliter/issues/4
+
+`twoliter update` is another inspiration from `cargo`, this time inspired by `cargo update`.
+This command makes it easy for maintainers to update the packages (kits actually) they are using to the latest versions.
+
+In the original Bottlerocket build system, we used Cargo crates to model RPM dependencies.
+In Twoliter, we will again lean on Cargo to manage dependencies for us.
+We will create a local Cargo registry to represent kit and SDK versions, then use `cargo update` to resolve the latest dependencies.
+
+Before describing this procedure, we should consider that no two versions of the same kit can be in the dependency tree.
```

Comment:
This is potentially solved by introducing the same tight version binding between kits as exists between SDK and kit.